### ISSUE TITLE
With dzil run command -x  flags are eaten by dzil

### ISSUE THREAD

**Issue by ambs** [rjbs/Dist-Zilla#632]
`dzil run prove -v -b t/some-test` vs `dzil run 'prove -v -b t/some-test'`
In the first, -v and -b are interpreted as flags for dzil, and it shouldn't (or at least, it would be nice to).

**Comment by karenetheridge** [comment]
This is pretty standard for shell things -- try `dzil run -- prove -v -b t/some-test`.

**Comment by ambs** [comment]
True. I just though it would be good to report and suggest. Happy if you wish to just close the issue.

**Comment by jonasbn** [comment]
I can confirm the observation by @ambs

**Comment by jonasbn** [comment]
Commented lifted from PR #635 

> @Leont I understand your standpoint, the suggested fix is not pretty. Could you please elaborate on what you mean by permutation and how you would approach eliminating the use of --, since it is a common usage pattern.

**Comment by Leont** [comment]
'permute' is the getopt option that allows it to not stop at the first non-option (in this case 'prove'), but skip over it and only consume the options (and their arguments).

What I don't understand is how the getopt doesn't consume the `--`. `App::Cmd` does some pretty complex things to parse arguments.
